api/lib/infrastructure/repositories/user-repository.js

```diff
@@ -286,6 +285,19 @@ module.exports = {
       });
   },
 
+  async isEmailAllowedToUseForCurrentUser(userId, email) {
```

Comment:
```suggestion
  async canUserUsemail(userId, email) {
```
Pourquoi pas le nommer isEmailRegistered, et lever l'erreur dans le use-case ?
Là, on met des RG métiers dans les repo, et c'est pas très Clean Archi ?

Reply:
Pour le nommage, je me suis inspiré du format actuel isEmail... , dans notre standard de code, on lève bien des exceptions qui sont bien wrap par l'error manager. On peut en discuter au point tech pour avoir l'avis de la team